Added GitCMS docs in CMS section (#11235)
withastro/docs: src/content/docs/en/guides/cms/gitcms.mdx

---
title: GitCMS & Astro
description: Integrate GitCMS into your Astro project for seamless content management
sidebar:
  label: GitCMS
type: cms
service: GitCMS
i18nReady: true
---

[GitCMS](https://gitcms.blog) is a Chrome extension that transforms GitHub into a powerful headless CMS, providing a Notion-like editing experience for managing static sites built with frameworks like Astro, Next.js, Hugo, Jekyll, and more. It allows developers, clients, and content writers to manage content directly within GitHub's interface without additional servers or complex CMS setups.

> 
> ![GitCMS Notion-like editor](https://gitcms.blog/images/gitcms-editor-for-astro.webp)

Reviewer: Except in certain rare cases, we don't use images in the docs, and this is even more true when they are purely decorative. Everything should be described in words to be accessible to everyone. So I think we should remove this image:
```suggestion
```

## Prerequisites
- An existing Astro project with a GitHub repository.

> - [GitCMS Chrome Extension](https://gitcms.blog) installed.

Reviewer: I don't think this is a prerequisite since the purpose of the guide is to explain how to set up GitCMS, and this is indeed the first step of `Setting Up GitCMS`. Instead, perhaps we should say that any Chrome-based browser is required? (unless there is an extension for other browsers?)
```suggestion
- A Chrome-based browser installed.
```

## Setting Up GitCMS

> 1. **Install the GitCMS Chrome Extension**: Add the [GitCMS extension](https://gitcms.blog) to Chrome.
> 2. **Initialize GitCMS in Your Repository**:
>    - Open your GitHub repository.
>    - Click the GitCMS icon in the Chrome extension menu.
>    - Click "Initialize."
>    - This redirects you to a new GitHub file creation page with `gitcms.md`, where the GitCMS configuration editor appears.

Reviewer: A few things here:
* I feel like we can flatten the list
* Maybe we should use the extension link directly to save an unnecessary click to the user
* It seems the button is named "Init" and not "Initialize"
* It seems there is one more steps before configuring (free/paid plan)
* Also, I don't know if it's my setup, but I had to reload the page to see the configuration editor, otherwise I had an empty .md file without any fields...
* I think we should show the complete process, so "Save & Commit" as well, because we can't assume the user's level and a beginner might not know how to save his configuration.
```suggestion

1. Install the [GitCMS Chrome Extension](https://chromewebstore.google.com/detail/gitcms-headless-cms-for-y/pemglpclheejlbobifgdbjjofdhmiafm).
2. Navigate to the Github repository containing your Astro project.
3. In the Chrome Extension menu, click on GitCMS to open the extension.
4. Click on **Init**. You will be redirected to an edit page to create a `gitcms.md` file. If the GitCMS configuration UI is not displayed, you may need to navigate between **Edit**/**Preview** modes or reload the page.
5. Click **Try for Free** or enter your license key if you choose to use a paid plan.
6. Configure GitCMS:
  * **Website URL**: Enter your site's URL (e.g., `https://yourblog.com`).
  * **Repository**: Your Github repository. This value is filled automatically and you should not need to change it.
  * **Media Files Folder**: Specify where media files are stored (e.g., `/public`).
7. Use the **Save & Commit** button to save your configuration. You should now have a `gitcms.md` file in the root of your repository.
```
Does this seem correct to you?

## Configuring GitCMS

> 1. **Website URL**: Enter your site's URL (e.g., `https://yourblog.com`).
> 2. **Media Folder**: Specify where media files are stored (usually managed by your static site generator).

Reviewer: Since I moved these fields in the previous step, perhaps we could explain how to change the configuration instead.
```suggestion

You can update the GitCMS configuration at any time by opening the `gitcms.md` file in Github. The GitCMS UI should be displayed automatically. Then, click on **Try for Free** or enter your license key to access to your configuration. The interface will allow you to update existing configuration or add new collections.
```

### Adding Collections
Collections group related content, such as blog posts, pages, or team members.

> 1. **Create a Collection**:
>    - Click "Add Collection."
>    - Name the collection (e.g., `posts`, `pages`, `authors`).
>    - Specify the folder where markdown files are stored (e.g., `src/content/posts`).
> 
> 2. **Define Frontmatter Schema**:
>    Configure metadata fields using:
>    - **Title Field**: For post titles.
>    - **Single Line Field**: For short text like descriptions.
>    - **Multi Line Field**: For longer content.
>    - **DateTime**: For publishing dates.
>    - **MultiSelect**: For categories and tags.
>    - **File Input**: For media uploads.
>    - **Boolean Field**: For toggles like "draft."

Reviewer: Same here, my suggestions are mostly formatting related:
* point 1 could be flattened I think
* even though "Filename Pattern" is optional, I think we should display it in the list
* point 2 has a formatting issue because Markdown doesn't treat a single line break as a line break and perhaps we should use the actual field names
```suggestion
To create a new collection:

1. Click **Add Collection**
2. Define the **Collection Settings**:
   - **Collection Name**: An unique name for your collection (e.g., `posts`, `pages`, `authors`).
   - **Collection Folder**: The path where your collection is stored (e.g., `/src/data/authors`)
   - **Filename Pattern**: Defines a template for the names of files in this collection by reusing fields from frontmatter (optional)
3. Define the **Frontmatter Fields** by associating a name with a type. The available types are:
   - **title**: For entry titles.
   - **singleline**: For short text like descriptions.
   - **multiline**: For longer content.
   - **datetime**: For dates (e.g. publishing date).
   - **multi_select**: For categories and tags.
   - **file**: For media uploads.
   - **boolean**: For toggles like "draft".
   
You can add as many collections as you like. Then, once you're done editing, click **Save & Commit** to update your configuration.
```

Perhaps we need one more step or explain that the "Collection Folder" must be an existing folder/must be created manually. I mean, I can save the config with a path that doesn't exist yet. But then, to create a new entry, I get a 404 because the folder doesn't exist. Is this correct or is there a method I haven't seen?

Author: Yeah, it will redirected to GitHub's 404 page. I think that is what intended. But yeah we could also mention this edge case to readers that collection folder path must exist.

>    Example configuration:

Reviewer: ```suggestion
Here is an example configuration that defines a `posts` collection:
```

   ```yaml
   ---
   media_path: /public
   repo: YourUsername/your-repo
   website_url: https://yourwebsite.com
   
   collections:
     - name: posts
       file_extension: .md
       filename_pattern: title
       git_path: /src/content/posts
       metadata_schema:
         - name: title
           required: true
           type: title
         - name: description
           required: false
           type: multiline
         - name: date
           required: true
           type: datetime
         - name: image
           required: false
           type: file
         - allowed_values:
             - Product Updates
             - Features
             - Comparison
           name: categories
           required: true
           type: multi_select
         - allowed_values:
             - AuthorName
           name: authors
           required: true
           type: multi_select
         - name: draft
           required: false
           type: boolean
   ---
   ```
   GitCMS generates this automatically, so manual editing is unnecessary.

> ## Using GitCMS

Reviewer: I haven't edited this section, but maybe this should be a bit more detailed? Here are some ideas:
* Perhaps a link to https://gitcms.blog/posts/how-to-configure-gitcms/#understanding-media-folder-structure could be useful especially if someone try to add GitCMS to an existing project (with existing posts and images)
* The GitCMS button to create a new entry instead of creating a file manually with Github interface
* The Markdown content is divided in blocks and that the user must use the "+" button to add text, headings, etc.

1. **Open the GitCMS Editor**:

>    - Navigate to a markdown file in your repository.

Reviewer: nit:
```suggestion
   - Navigate to a Markdown file in your repository.
```

   - Click the GitCMS icon to toggle between GitHub and GitCMS editors.

2. **Editing Features**:

>    - Visual markdown editing.

Reviewer: nit:
```suggestion
   - Visual Markdown editing.
```

   - Structured metadata forms.
   - Media uploads with previews.

## Rendering Content in Astro
Use Astro's Content Collections API to display content.

### Displaying a Collection List

> ```tsx

Reviewer: We use `astro` for Astro files:
````suggestion
```astro
````

---
import { getCollection } from 'astro:content';
const posts = await getCollection('posts');
---
<ul>
  {posts.map(post => (
    <li>

>       <a href={`/posts/{post.slug}`}>{post.data.title}</a>

Reviewer: Not sure, but I guess with Astro 5 this should be:
```suggestion
      <a href={`/posts/{post.id}`}>{post.data.title}</a>
```

    </li>
  ))}
</ul>
```

### Displaying a Single Entry

> ```tsx

Reviewer: We use `astro` for Astro files:
````suggestion
```astro
````

---

> import { getEntry } from 'astro:content';
> 
> const post = await getEntry('posts', 'my-first-post');
> const { Content } = await post.render();

Reviewer: In Astro 5 this should be:
```suggestion
import { getEntry, render } from 'astro:content';

const post = await getEntry('posts', 'my-first-post');
const { Content } = await render(post);
```

---
<main>
  <h1>{post.data.title}</h1>
  <Content />
</main>
```

> For more on querying content, see Astro's [Content Collections documentation](https://docs.astro.build/en/guides/content-collections/).

Reviewer: Fix link (we don't include the domain name):
```suggestion
For more on querying content, see Astro's [Content Collections documentation](/en/guides/content-collections/).
```

## Deploying with Astro

> Refer to Astro's [deployment guides](https://docs.astro.build/en/guides/deploy/) for hosting options. Since GitCMS stores content in GitHub, deployment remains simple.

Reviewer: Fix link (we don't include the domain name):
```suggestion
Refer to Astro's [deployment guides](/en/guides/deploy/) for hosting options. Since GitCMS stores content in GitHub, deployment remains simple.
```

## Resources
- [Introducing GitCMS](https://gitcms.blog/posts/introducing-gitcms/)
- [How to Configure GitCMS](https://gitcms.blog/posts/how-to-configure-gitcms/)
- [Frontmatter Fields in GitCMS](https://gitcms.blog/posts/frontmatter-editor/)
- [GitCMS Chrome Extension](https://gitcms.blog)

> 
> By integrating GitCMS with Astro, developers and content creators gain a seamless, efficient, and user-friendly content management experience.

Reviewer: This line doesn't seem to be related to "Resources" and is more or less what we already explain at the beginning of the guide, right? So perhaps we could delete it:
```suggestion
```

Reviewer 2: Yes, not appropriate here.

---
title: GitCMS & Astro
description: Integrate GitCMS into your Astro project for seamless content management
sidebar:
  label: GitCMS
type: cms
service: GitCMS
i18nReady: true
---

> [GitCMS](https://gitcms.blog) is a Chrome extension that bridges the gap in static site generation (SSG) workflows by enhancing the writing and management experience for markdown files. It transforms GitHub into a powerful headless CMS, providing a Notion-like editing experience directly within GitHub's interface. With GitCMS, developers, clients, and content writers can write, manage, and search markdown content effortlessly — eliminating the pain of editing raw markdown or juggling file management. This makes managing content for Astro sites smoother, without additional servers or complex CMS setups.

Reviewer 2: ```suggestion
[GitCMS](https://gitcms.blog) is a Chrome extension that enhances GitHub's Markdown editing interface, creating a Git-based, headless CMS in your browser.
```
These introductions are always short, and factual. Not marketing-speak. Something like this is what we're going for. I believe this seems accurate based on what I researched, but you can suggest an alternative!

> ## Resources

Reviewer 2: ```suggestion
## Official Resources
```
We distinguish here, in case people ever come to add community resources like blog posts about using the CMS. (In which case, we'll add a ## Comunity Resources heading, too!)

- [Introducing GitCMS](https://gitcms.blog/posts/introducing-gitcms/)

> - [How to Configure GitCMS](https://gitcms.blog/posts/how-to-configure-gitcms/)
> - [Frontmatter Fields in GitCMS](https://gitcms.blog/posts/frontmatter-editor/)

Reviewer 2: ```suggestion
```
We don't typically include multiple resources from the same official source. We don't need a link to each, individual topic. A link to overall documentation (or, in your case an introduction) and to the Chrome extension in the web store is fine! People can find what they need!
